components/eam/src/chemistry/utils/modal_aero_wateruptake.F90

```diff
@@ -299,206 +251,228 @@ subroutine modal_aero_wateruptake_dr(state, pbuf, list_idx_in, dgnumdry_m, dgnum
+      enddo ! kk
 
-            dgncur_awet(i,k,m) = dgncur_a(i,k,m) * (wetrad(i,k,m)/dryrad(i,k,m))
-            qaerwat(i,k,m)     = rhoh2o*naer(i,k,m)*wtrvol(i,k,m)
-
-            ! compute aerosol wet density (kg/m3)
-            if(compute_wetdens) then
-               if (wetvol(i,k,m) > 1.0e-30_r8) then
-                  wetdens(i,k,m) = (drymass(i,k,m) + rhoh2o*wtrvol(i,k,m))/wetvol(i,k,m)
-               else
-                  wetdens(i,k,m) = specdens_1(m)
-               end if
-            endif
-         end do ! i = 1, ncol
-      end do ! k = top_lev, pver
+   enddo    ! modes
 
-   end do ! m = 1, nmodes
+end subroutine modal_aero_wateruptake_dryaer
 
-   !----------------------------------------------------------------------------
-   ! write history output if not in diagnostic mode
+!===============================================================================
+subroutine modal_aero_wateruptake_rh_clearair( ncol,         & ! in
+                     temperature,    pmid,   h2ommr,  cldn,  & ! in
+                     rh                                      ) ! out
+!-----------------------------------------------------------------------
+! estimate clear air relative humidity using cloud fraction
+!-----------------------------------------------------------------------
+use wv_saturation,    only: qsat_water
 
-   if (.not.present(list_idx_in)) then
+   integer,  intent(in) :: ncol                 ! number of columns
+   real(r8), intent(in) :: temperature(:,:)     ! temperature [K]
+   real(r8), intent(in) :: pmid(:,:)            ! layer pressure [Pa]
+   real(r8), intent(in) :: h2ommr(:,:)          ! water mass mixing ratio [kg/kg]
+   real(r8), intent(in) :: cldn(:,:)            ! layer cloud fraction [fraction]
+   real(r8), intent(out) :: rh(:,:)           ! relative humidity [Fraction]
 
-      aerosol_water(:ncol,:) = 0._r8
-      do m = 1, nmodes
-         ! output to history
-         write( trnum, '(i3.3)' ) m
-         call outfld( 'wat_a'//trnum(3:3),  qaerwat(:,:,m),     pcols, lchnk)
-         call outfld( 'dgnd_a'//trnum(2:3), dgncur_a(:,:,m),    pcols, lchnk)
-         call outfld( 'dgnw_a'//trnum(2:3), dgncur_awet(:,:,m), pcols, lchnk)
-         if (history_aerosol .and. .not. history_verbose) &
-         aerosol_water(:ncol,:) = aerosol_water(:ncol,:) + qaerwat(:ncol,:,m)
-      end do ! m = 1, nmodes
-
-      if (history_aerosol .and. .not. history_verbose) &
-         call outfld( 'aero_water',  aerosol_water(:ncol,:),    ncol, lchnk)
+   ! local variables
+   integer   :: icol, kk
+   real(r8)  :: rh_max = 0.98_r8        ! maximum relative humidity
+   real(r8)  :: es(pcols)               ! saturation vapor pressure [Pa]
+   real(r8)  :: qs(pcols)               ! saturation specific humidity [kg/kg]
+   real(r8), parameter  :: cldn_thresh = 1.0_r8      ! threshold cloud fraction
+
+   do kk = top_lev, pver
+      call qsat_water(temperature(:ncol,kk), pmid(:ncol,kk), es(:ncol), qs(:ncol))
```

Comment:
I guess `es` is never used, so the only output that is used is `qs`. We still need to send `es` as it is not an optional variable.

Reply:
Yes. I'll keep `es `as it is a required output variable of `qsat_water`.